### PR TITLE
[pkg/ottl] Add GetXML Converter

### PR DESCRIPTION
**Pull request by djaglowski**
This Converter simply allows users to query an XML document and return an XML document composed of the selected elements. Along with [`RemoveXML`](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35301) and [`InsertXML`](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35436), I think this gives us a decently robust set of XML-flavored crud operations.

### CODE REVIEW

File: pkg/ottl/ottlfuncs/func_get_xml.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	if err := validateXPath(args.XPath); err != nil {
+		return nil, err
+	}
+
+	return getXML(args.Target, args.XPath), nil
+}
+
+// getXML returns a XML formatted string that is a result of removing all matching nodes from the target XML.
+// This currently supports removal of elements, attributes, text values, comments, and CharData.
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_get_xml.go:39]: This line seems to contradict the line in the README that claims only selecting is supported.

File: pkg/ottl/ottlfuncs/func_get_xml.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		}
+
+		result := &xmlquery.Node{Type: xmlquery.DocumentNode}
+		for _, n := range nodes {
+			if n.Type != xmlquery.ElementNode {
+				continue
+			}
+			xmlquery.AddChild(result, n)
+		}
+		return result.OutputXML(false), nil
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_get_xml.go:61]: parsing from a string and returning a string gives use the most flexibility in how the converters are used, but I am a little worried about performance: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35281#issuecomment-2379644212

File: pkg/ottl/ottlfuncs/func_remove_xml.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ func parseNodesXML(targetVal string) (*xmlquery.Node, error) {
 	if err != nil {
 		return nil, fmt.Errorf("parse xml: %w", err)
 	}
-	if !preserveDeclearation {
+	if !preserveDeclearation && top.FirstChild != nil {
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_remove_xml.go:85]: Is this change related?

Comment (djaglowski) [reply pkg/ottl/ottlfuncs/func_remove_xml.go:85]: Not strictly, but it was noticed while adding tests for this. I'll move it over to #35463.

File: pkg/ottl/ottlfuncs/func_get_xml.go (hunk 1 above)
Comment (djaglowski) [reply pkg/ottl/ottlfuncs/func_get_xml.go]: Good catch. Just a copy/paste error. Updated now.

**Review by TylerHelmuth — APPROVED** [review]
